src/plugins/es_ui_shared/public/request/use_request.test.ts

```diff
@@ -117,6 +117,18 @@ describe('useRequest hook', () => {
         expect(hookResult.error).toBe(getErrorResponse().error);
       });
 
+      it('applies errorInterceptors to errors', async () => {
+        const { setupErrorRequest, completeRequest, hookResult, getErrorResponse } = helpers;
+        const errorInterceptors = [
+          (error: any) => ['Error is:', error.statusText],
```

Comment:
The functionality described in the PR description can be accomplish by creating side effects within an error interceptor, e.g. a call to `setState`.